pkg/apis/application/v1alpha1/types.go

```diff
@@ -541,6 +546,26 @@ func (spec ApplicationSpec) GetProject() string {
 	return spec.Project
 }
 
+func (proj AppProject) IsDefault() bool {
+	return proj.Name == "" || proj.Name == common.DefaultAppProjectName
+}
+
+func (proj AppProject) IsClusterGroupKindPermitted(gk metav1.GroupKind) bool {
+	if proj.IsDefault() {
+		return true
+	}
```

Comment:
Actually we no longer treat the default project as anything special, other than it cannot be deleted. So we should remove any special consideration for the default project.

Reply:
Got catch! Removed special treatment and added default cluster resource whitelist to the default project.